src/dotnet-interactive-vscode/src/vscode/commands.ts

```diff
@@ -121,6 +122,32 @@ export function registerFileFormatCommands(context: vscode.ExtensionContext, cli
         '.NET Interactive Notebooks': ['dib', 'dotnet-interactive']
     };
 
+    function workspaceHasUnsavedNotebookWithName(fileName: string): boolean {
+        return vscode.workspace.textDocuments.findIndex(textDocument => {
+            if (textDocument.notebook) {
+                const notebookUri = textDocument.notebook.uri;
+                return isUnsavedNotebook(notebookUri) && path.basename(notebookUri.fsPath) === fileName;
+            }
+
+            return false;
+        }) >= 0;
+    }
+
+    function getNewNotebookName(): string {
+        let suffix = 1;
+        while (workspaceHasUnsavedNotebookWithName(`Untitled-${suffix}.dib`)) {
```

Comment:
do we need to add the .dib ? Untitled-notebook-1 looks good enough and we leave the choice to the user at save time

Reply:
Yes, we need to specify `.dib`.  A few lines down that's the part that makes it the default extension when the user finally tries to save.